static/js/gpio-control.js

```diff
@@ -0,0 +1,35 @@
+document.addEventListener('DOMContentLoaded', () => {
+    const gpioButton = document.getElementById('gpioButton');
+    const ledIndicator = document.getElementById('ledIndicator');
+
+    // Button click handler
+    gpioButton.addEventListener('click', async () => {
+        try {
+            const response = await fetch('/api/gpio/led/toggle', {
+                method: 'POST',
+            });
+            const data = await response.json();
+
+            if (data.status === 'success') {
+                updateLedIndicator(data.led_state);
+                alert(data.state.message || 'LED state toggled'); // Show alert popup with message
```

Comment:
@Dillon: With the data you get from Flask through `fetch('/api/gpio/led/toggle',` that get data from `def toggle_led():` you will trigger alter here.